### PR TITLE
533-fe-manage-users-modal

### PR DESCRIPTION
**Pull request by JuditKaramazov**
**SettingsManager changes**:

- [x] It doesn't show the Usuaris tab if the user doesn't have enough permissions (role: mentor or higher).

- [x] Only admin roles can edit the users.

- [x] Since React Fragments and a direct import of AccountAdmin were causing problems in the application, some relevant changes (like importing AccountAdmin) have been made by updating UsersManager.tsx.

- [x] The type of Tabs.tsx has been extended in order to avoid duplicates.

- [x] Tests have been updated accordingly.

### CODE REVIEW

File: front/src/components/organisms/SettingsManager.tsx [hunk 1]
```diff
@@ -21,12 +22,29 @@ const tabsData: TTabsData[] = [
...
-  const tTabsData = tabsData.map((tab) => ({
-    title: t(tab.title),
-    tabComponent: tab.tabComponent,
-  }))
+  const tTabsData = tabsData
+    .filter((tab) => {
+      if (tab.title === 'Usuarios') {
+        return user?.role === 'MENTOR' || user?.role === 'ADMIN'
+      }
+      return true
```
Comment (kevinmamaqi) [inline front/src/components/organisms/SettingsManager.tsx:34]:
```suggestion
      return false
```

**Review by kevinmamaqi — CHANGES_REQUESTED** [review]
Add `id` and `requiredRole`.

File: front/src/components/organisms/SettingsManager.tsx [hunk 2]
```diff
@@ -3,30 +3,49 @@ import { useTranslation } from 'react-i18next'
 import { TopicsManagerBoard } from './TopicsManagerBoard'
 import { UsersManager } from './UsersManager'
 import { Tabs } from '../molecules/Tabs'
+import { useAuth } from '../../context/AuthProvider'
 
 type TTabsData = {
```
Comment (kevinmamaqi) [inline front/src/components/organisms/SettingsManager.tsx:8]: Extender el type de Tabs.tsx para evitar duplicidades.

**Review by kevinmamaqi — CHANGES_REQUESTED** [review]

**Review by kevinmamaqi — APPROVED** [review]

File: front/src/__tests__/organisms/SettingsManager.test.tsx [hunk 3]
```diff
@@ -1,33 +1,144 @@
...
+  it('renders Temes and Usuaris tabs correctly for mentor roles or higher', () => {
     render(<SettingsManager />)
 
     expect(screen.getByText('Temes')).toBeInTheDocument()
-    expect(screen.getByText(/usuaris/i)).toBeInTheDocument()
-    expect(
-      screen.getByText(
-        /No hi ha temes disponibles. *Accedeix des d'una categoria per veure o gestionar els temes./
-      )
-    ).toBeInTheDocument()
```
Comment (NataliaR-BCN) [inline front/src/__tests__/organisms/SettingsManager.test.tsx:14]: This text is still available, isn't it? 
Or what happens in the Temes tab if a Mentor/the Admin opens the SettingsManager from Home page, where there is not slug (needed to render topics)?

File: front/src/__tests__/organisms/SettingsManager.test.tsx [hunk 4]
```diff
@@ -1,33 +1,144 @@
...
+      expect(screen.getByText('Usuaris')).toBeInTheDocument()
+    })
+  
+    expect(screen.queryByText(/No hi ha temes disponibles./)).not.toBeInTheDocument()
+    
+    fireEvent.click(screen.getByText('Temes'))
+  
+    await waitFor(() => {
+      expect(screen.getByText('Temes')).toBeInTheDocument()
+    })
```
Comment (NataliaR-BCN) [inline front/src/__tests__/organisms/SettingsManager.test.tsx:73]: You're expecting a text that you already clicked

File: front/src/__tests__/organisms/SettingsManager.test.tsx [hunk 5]
```diff
@@ -1,33 +1,144 @@
...
+  })
+
+  it('changes content tab according to click on menu tab', async () => {
+    render(<SettingsManager />)
+  
+    fireEvent.click(screen.getByText('Usuaris'))
+  
+    await waitFor(() => {
+      expect(screen.getByText('Usuaris')).toBeInTheDocument()
+    })
```
Comment (NataliaR-BCN) [inline front/src/__tests__/organisms/SettingsManager.test.tsx:65]: You're expecting a text that you already clicked

File: front/src/__tests__/organisms/UsersManager.test.tsx [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+    const user2 = screen.getByText('User Two')
+    expect(user2).toBeInTheDocument()
+
+    const searchInput = screen.getByPlaceholderText('Escribe el DNI')
+    fireEvent.change(searchInput, { target: { value: '12345678' } })
+
+    await waitFor(() => {
+      expect(screen.queryByText('User Two')).toBeInTheDocument()
+    })
+
```
Comment (NataliaR-BCN) [inline front/src/__tests__/organisms/UsersManager.test.tsx:82]: 'User Two' already expected 6 lines ago

File: front/src/components/organisms/UsersManager.tsx [hunk 7]
```diff
@@ -1 +1,15 @@
-export const UsersManager = () => <>Users Manager</>
+import { useAuth } from '../../context/AuthProvider'
+import { AccountAdmin } from '../../pages/AccountAdim'
+
+const UsersManager = () => {
+  const { user } = useAuth()
+
+  return (
+    <>
+      Users Manager
```
Comment (NataliaR-BCN) [inline front/src/components/organisms/UsersManager.tsx:9]: Translate this text, please!

**Review by kevinmamaqi — APPROVED** [review]

File: front/src/__tests__/organisms/SettingsManager.test.tsx [hunk 8]
```diff
@@ -15,19 +61,82 @@ describe('SettingsManager component', () => {
...
+    
+    fireEvent.click(screen.getByText('Temes'))
+  
+    expect(screen.queryByText('Users Manager')).not.toBeInTheDocument()
+  })
+})
 
-    fireEvent.click(screen.getByText(/usuaris/i))
+describe('User Permissions', () => {
+  it('only allows admins to fetch users, and renders an error if the process fails', async () => {
```
Comment (kevinmamaqi) [inline front/src/__tests__/organisms/SettingsManager.test.tsx:78]: Corregir test, ¿Como sabemos si el proceso falla?

File: front/src/__tests__/organisms/SettingsManager.test.tsx [hunk 9]
```diff
@@ -15,19 +61,82 @@ describe('SettingsManager component', () => {
     expect(screen.queryByText('Users Manager')).not.toBeInTheDocument()
   })
 
-  it('should change content tab according to click on menu tab', async () => {
+  it('changes content tab according to click on menu tab', async () => {
```
Comment (kevinmamaqi) [inline front/src/__tests__/organisms/SettingsManager.test.tsx:64]: No parece necesario que sea asincrono

**Review by kevinmamaqi — CHANGES_REQUESTED** [review]
Revisar tests y utilizar msw (mock service worker) y sus handlers.

**Comment by NataliaR-BCN** [comment]
As AccountAdmin is now rendered from another component, maybe it should be in components instead of pages, and deleted from router in main

**Review by kevinmamaqi — APPROVED** [review]

**Review by ViuCoding — APPROVED** [review]

**Pull request merged**